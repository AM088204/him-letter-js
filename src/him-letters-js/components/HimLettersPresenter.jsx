import React from 'react';
import Tabs from 'terra-tabs';
import TabContentTemplate from 'terra-tabs/lib/terra-dev-site/doc/example/TabContentTemplate';
import { injectIntl } from 'react-intl';
import HimSearchTemplate from './LetterTemplates/HimSearchTemplate';
import HimAddNewTemplateButton from './LetterTemplates/HimAddNewTemplateButton';
import '../stylesheets/HimLetterTemplates.css';
import Grid from 'terra-grid';
//Container to present all the components
const HimLettersPresenter = (props) => {

  const { intl } = props;

  const distributionLettersTab = (
    <Tabs.Pane
      label={intl.formatMessage({ id: 'him-letters-engine.distributeLettersTabLabel' })}
      key="DistributeLetters">
      <TabContentTemplate label={intl.formatMessage({ id: 'him-letters-engine.distributeLettersTabLabel' })}>
        {/* to do */}
      </TabContentTemplate>
    </Tabs.Pane>
  );

  const lettersTemplatesTab = (
    <Tabs.Pane
      label={intl.formatMessage({ id: "him-letters-engine.letterTemplateTabLabel" })}
      key="LetterTemplates">
      <TabContentTemplate label={intl.formatMessage({ id: "him-letters-engine.letterTemplateTabLabel" })}>
        <div className='align-left'>
            <HimSearchTemplate />
          </div>
          <div className='align'>
            <HimAddNewTemplateButton disclosureType="modal" size="large" />
          </div>
      </TabContentTemplate>
    </Tabs.Pane>
  );
  return (
    <div>
      <Tabs>
        {distributionLettersTab}
        {lettersTemplatesTab}
      </Tabs>
    </div>
  );
};

export default injectIntl(HimLettersPresenter);