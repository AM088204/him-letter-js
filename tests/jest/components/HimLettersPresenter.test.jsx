import React from 'react';
import { shallowWithIntl, mountWithIntl } from 'terra-enzyme-intl';
import TabContentTemplate from 'terra-tabs/lib/terra-dev-site/doc/example/TabContentTemplate';
import  HimLettersPresenter from '../../../src/him-letters-js/components/HimLettersPresenter';

describe('HimLettersPresenter', () => {
    it('Should Render HimLettersPresenter', () => {
        const himLettersPresenter = shallowWithIntl(<HimLettersPresenter />);
        expect(himLettersPresenter.first().shallow()).toMatchSnapshot();
    });

    it('should verify the label for tabs', () => {
        const himLettersPresenter = mountWithIntl(
            <HimLettersPresenter />,
        );
        expect(himLettersPresenter.find(TabContentTemplate).props().label).toStrictEqual('him-letters-engine.distributeLettersTabLabel','him-letters-engine.letterTemplateTabLabel');
    });

}); 