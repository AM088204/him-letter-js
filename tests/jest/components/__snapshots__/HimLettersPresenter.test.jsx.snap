// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HimLettersPresenter Should Render HimLettersPresenter 1`] = `
<div>
  <Tabs
    fill={false}
    responsiveTo="parent"
    tabFill={false}
  >
    <TabPane
      isActive={false}
      isDisabled={false}
      isIconOnly={false}
      key="DistributeLetters"
      label="him-letters-engine.distributeLettersTabLabel"
    >
      <TabContentTemplate
        label="him-letters-engine.distributeLettersTabLabel"
      />
    </TabPane>
    <TabPane
      isActive={false}
      isDisabled={false}
      isIconOnly={false}
      key="LetterTemplates"
      label="him-letters-engine.letterTemplateTabLabel"
    >
      <TabContentTemplate
        label="him-letters-engine.letterTemplateTabLabel"
      >
        <div
          className="align-left"
        >
          <InjectIntl(HimSearchTemplate) />
        </div>
        <div
          className="align"
        >
          <InjectIntl(HimAddNewTemplateButton)
            disclosureType="modal"
            size="large"
          />
        </div>
      </TabContentTemplate>
    </TabPane>
  </Tabs>
</div>
`;
